src/printer.js

```diff
@@ -3097,6 +3084,40 @@ function printFunctionParams(path, print, options, expandArg, printTypeParams) {
   ]);
 }
 
+function shouldPrintParamsWithoutParens(path, options) {
+  if (options.arrowFnParens === "always") {
+    return false;
+  }
+
+  const node = path.getValue();
+  if (options.arrowFnParens === "avoid") {
+    return canPrintParamsWithoutParens(node);
+  }
+
+  // "default" mode; omit parens only when:
+  // - currying (eg; fn = a => b => c)
+  // - used as a sole callback (eg; arr.map(x => x + 1))
+  // - used in template literals (eg; `color: ${props => props.color}`) for styled-components et al
+  if (options.arrowFnParens === "default") {
+    const parentNode = path.getParentNode();
+    if (
+      (parentNode.type === "CallExpression" &&
+        (parentNode.arguments.length === 1 ||
+          (parentNode.arguments[0] === node &&
+            !parentNode.arguments.some(
+              (n, i) => i > 0 && n.type === "ArrowFunctionExpression"
```

Comment:
Maybe check for `"FunctionExpression"` too?

Reply:
that would not fit with the currying style.